pyfar/classes/filter.py

```diff
@@ -446,35 +448,55 @@ def init_state(self, cshape, state='zeros'):
                 new_state[idx, ...] = spsignal.lfilter_zi(coeff[0], coeff[1])
         super().init_state(state=new_state)
 
-    def minimum_impulse_response_length(self, unit='samples'):
-        """
+    def minimum_impulse_response_length(self, unit='samples', tolerance=None):
```

Comment:
we could also think about setting this to something like 2*eps, so numerical errors are always accounted for.

Reply:
even better!